src/seedu/addressbook/data/person/Address.java

```diff
@@ -8,39 +8,81 @@
  */
 public class Address {
 
-    public static final String EXAMPLE = "123, some street";
-    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
-    public static final String ADDRESS_VALIDATION_REGEX = ".+";
+    public static final String EXAMPLE = "a/123, Clementi Ave 3, #12-34, 231534";
+    public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses must be in "
+            + "the a/BLOCK, STREET, UNIT, POSTAL_CODE format";
+    
+    private static final int BLOCK_INDEX = 0;
+    private static final int STREET_INDEX = 1;
+    private static final int UNIT_INDEX = 2;
+    private static final int POSTAL_INDEX = 3;
+    
+    private static final int ADDRESS_COMPONENTS = 4;
+    
+    private static final String SPLIT_REGEX = ",";
 
     public final String value;
     private boolean isPrivate;
 
+    private final Block block;
+    private final Street street;
+    private final Unit unit;
+    private final PostalCode postalCode;
+    
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        String trimmedAddress = address.trim();
+        String trimmedAddress = address.trim();             
+ 
         this.isPrivate = isPrivate;
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        
+        String blockNo = address.split(SPLIT_REGEX)[BLOCK_INDEX];
+        String streetName = address.split(SPLIT_REGEX)[STREET_INDEX];
```

Comment:
what if some one doesn't add a comma?
what if there are lesser number of fields than the required?